service/src/main/java/bio/terra/workspace/service/resource/ResourceValidationUtils.java

```diff
@@ -441,14 +455,35 @@ public static void validateAzureVmSize(String vmSize) {
     }
   }
 
-  public static void validateRegion(String region) {
+  public static void validateAzureRegion(String region) {
+    if (StringUtils.isEmpty(region)) {
+      // Azure resources like workspaces may not have a region.
+      logger.warn("Cannot validate empty Azure region.");
+      return;
+    }
     if (!Region.values().stream()
         .map(Region::toString)
         .collect(Collectors.toList())
         .contains(region)) {
       logger.warn("Invalid Azure region {}", region);
-      throw new InvalidReferenceException(
-          "Invalid Azure Region specified. See the class `com.azure.core.management.Region`");
+      throw new InvalidControlledResourceException("Invalid Azure Region specified.");
+    }
+  }
+
+  public static void validateGcpRegion(
+      TpsApiDispatch tpsApiDispatch, UUID workspaceId, String region) {
+    region = GcpUtils.parseRegion(region);
+
+    // Old workspaces may not have a policy. We can only validate against policy if one exists.
+    if (tpsApiDispatch.getPaoIfExists(workspaceId).isPresent()) {
```

Comment:
Another change to verify we have a policy before checking regions against it.

Reply:
Should you have a `logger.warn` here also?

Reply:
Good idea